src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FirstLastN.cs

```diff
@@ -66,5 +66,32 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             return fArgsValid;
         }
     }
+
+    // FirstN(source:*, [count:n])
+    // LastN(source:*, [count:n])
+    internal sealed class FirstLastNFunction_UO : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        // Note this function does not inherit from FunctionWithTableInput so there cannot be a common
+        // base class with the above function
+        public FirstLastNFunction_UO(bool isFirst)
+            : base(isFirst ? "FirstN" : "LastN", isFirst ? TexlStrings.AboutFirstN : TexlStrings.AboutLastN, FunctionCategories.Table, DType.UntypedObject, 0, 1, 2, DType.UntypedObject, DType.Number)
```

Comment:
>1, [](http://example.com/codeflow?start=159&length=2)

We're going to make the second argument for FirstN/LastN required soon - we can already start the untyped object version with two required arguments.

Reply:
Done